src/main/java/com/google/devtools/build/lib/rules/java/JavaConfiguration.java

```diff
@@ -353,6 +355,9 @@ public boolean isTranslationsDisabled() {
       defaultLabel = "//tools/jdk:toolchain",
       defaultInToolRepository = true)
   public Label getToolchainLabel() {
+    if (useRemoteJavaToolchain) {
```

Comment:
If people explicitly specify a toolchain, they should get their selection. This is important since you want to turn this incompatible flag on by default in 0.24 or so. The current code would override this with the remote toolchain.